src/pet.js

```diff
@@ -0,0 +1,82 @@
+const MAXIMUM_FITNESS = 10;
+const MINIMUM_HUNGER = 0;
+
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = MINIMUM_HUNGER;
+    this.fitness = MAXIMUM_FITNESS;
+    this.children = [];
+}
+
+Pet.prototype = {
+    get isAlive() {
+        return this.fitness > 0 && this.hunger < 10 && this.age < 30;
+    }
+}
+
+Pet.prototype.growUp = function() {
+    if(!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+    }
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+};
+
+Pet.prototype.walk = function() {
+    if(!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+    }
+    if ((this.fitness + 4) <= MAXIMUM_FITNESS) {
+        this.fitness += 4;
+    } else {
+        this.fitness = MAXIMUM_FITNESS;
+    };
+};
+
+Pet.prototype.feed = function() {
+    if(!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+    }
+    if ((this.hunger - 3) >= MINIMUM_HUNGER) {
+        this.hunger -= 3;
+    } else {
+        this.hunger = MINIMUM_HUNGER;
+    };
```

Comment:
As above this could be refactored to be a bit more concise using the built in `Math` library